src/main/java/com/google/devtools/build/remote/client/ActionGrouping.java

```diff
@@ -21,19 +30,71 @@
 
   @VisibleForTesting static final String actionString = "Entries for action with hash '%s'\n";
 
-  // Key: actionId; Value: a set of associated log entries.
-  Map<String, Multiset<LogEntry>> actionMap = new HashMap<>();
-  int numSkipped = 0;
+  // A summary of ActionResult for a single action:
+  // This finds and records an ActionResult, regardless of how it was obtained.
+  static class ActionResultSummary {
```

Comment:
I would much rather you didn't have a separate class for finding an ActionResult associated with a given action id. You can achieve the same behavior by inlining one additional function in the ActionDetails, right? This way, you don't have two separate classes that accumulate/process data for a single action, which to me would be cleaner.